common-rest/src/test/java/org/uniprot/api/rest/controller/AbstractGetByIdsPostControllerIT.java

```diff
@@ -0,0 +1,339 @@
+
+        // then
+        response.andDo(log())
+                .andExpect(status().is(HttpStatus.BAD_REQUEST.value()))
+                .andExpect(header().doesNotExist("Content-Disposition"))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
+                .andExpect(jsonPath("$.messages.*", containsInAnyOrder("request body is missing")));
+    }
+
+    @Test
+    void getByIdsPostWithEmptyBodyFailure() throws Exception {
+        // when
+        ResultActions response =
+                getMockMvc()
+                        .perform(
+                                post(getGetByIdsPath())
+                                        .contentType(MediaType.APPLICATION_JSON)
+                                        .content("{}"));
+
+        // then
+        response.andDo(log())
+                .andExpect(status().is(HttpStatus.BAD_REQUEST.value()))
+                .andExpect(header().doesNotExist("Content-Disposition"))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
+                .andExpect(
+                        jsonPath(
+                                "$.messages.*",
+                                containsInAnyOrder(getErrorMessagesForEmptyJson())));
+    }
+
+    @Test
+    void getByIdsWithPassedFacetsIgnoredPostSuccess() throws Exception {
+        // when
+        ResultActions response =
+                getMockMvc()
+                        .perform(
+                                post(getGetByIdsPath())
+                                        .contentType(MediaType.APPLICATION_JSON)
+                                        .content(getJsonRequestBodyWithFacets()));
+
+        // then
+        response.andDo(log())
+                .andExpect(status().is(HttpStatus.OK.value()))
+                .andExpect(header().exists("Content-Disposition"))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
+                .andExpect(jsonPath("$.results.size()", is(10)))
+                .andExpect(jsonPath("$.facets").doesNotExist());
+    }
+
+    @Test
+    void getByIdsWithPassedFacetFilterIgnoredPostSuccess() throws Exception {
+        // when
+        ResultActions response =
+                getMockMvc()
+                        .perform(
+                                post(getGetByIdsPath())
+                                        .contentType(MediaType.APPLICATION_JSON)
+                                        .content(getJsonRequestBodyWithFacetFilter()));
+
+        // then
+        response.andDo(log())
+                .andExpect(status().is(HttpStatus.OK.value()))
+                .andExpect(header().exists("Content-Disposition"))
+                .andExpect(header().string(HttpHeaders.CONTENT_TYPE, APPLICATION_JSON_VALUE))
+                .andExpect(jsonPath("$.results.size()", is(10)))
+                .andExpect(jsonPath("$.facets").doesNotExist());
```

Comment:
We need to have facets and facetFilter when download=false

Reply:
download=false is a validation error. I had covered in test getByIdsBadRequestFailure. But i am going to add a specific test case where download=false